File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int coleman(string text);

int main(void)
{
    string text = get_string("Text: ");
    int grade = coleman(text);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}
int coleman(string text)
{
    int l = 0; int s = 0; int w = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            l++;
        }
        if (text[i] == ' ')
        {
            w++;
        }
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            s++;
        }
    }
    float index = 0.0588 * (l / (float) w) * 100 - 0.296 * (s / (float) w) * 100 - 15.8;
    return round(index);
}

File: sentimental-readability/readability.py
import math

def coleman(text):
    l = 0  # số lượng chữ cái
    s = 0  # số lượng câu
    w = 1  # số lượng từ (bắt đầu là 1 vì không có khoảng trắng đầu tiên)

    for char in text:
        if char.isalpha():
            l += 1
        if char == ' ':
            w += 1
        if char in ['.', '?', '!']:
            s += 1

    index = 0.0588 * (l / w) * 100 - 0.296 * (s / w) * 100 - 15.8
    return round(index)

def main():
    text = input("Text: ")
    grade = coleman(text)
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")

if __name__ == "__main__":
    main()
